grakn-core/src/main/java/ai/grakn/concept/ResourceType.java

```diff
@@ -169,12 +175,53 @@
-        public static final DataType<Integer> INTEGER = new DataType<>(Integer.class.getName(), Schema.ConceptProperty.VALUE_INTEGER);
-        public static final DataType<Long> LONG = new DataType<>(Long.class.getName(), Schema.ConceptProperty.VALUE_LONG);
-        public static final DataType<Double> DOUBLE = new DataType<>(Double.class.getName(), Schema.ConceptProperty.VALUE_DOUBLE);
-        public static final DataType<Float> FLOAT = new DataType<>(Float.class.getName(), Schema.ConceptProperty.VALUE_FLOAT);
+        public static final DataType<String> STRING = new DataType<>(
+                String.class.getName(),
+                Schema.ConceptProperty.VALUE_STRING,
+                (v) -> v,
+                o -> defaultConverter(o, String.class, Object::toString));
+
+        public static final DataType<Boolean> BOOLEAN = new DataType<>(
+                Boolean.class.getName(),
+                Schema.ConceptProperty.VALUE_BOOLEAN,
+                (v) -> v,
+                o -> defaultConverter(o, Boolean.class, (v) -> Boolean.parseBoolean(v.toString())));
+
+        public static final DataType<Integer> INTEGER = new DataType<>(
+                Integer.class.getName(),
+                Schema.ConceptProperty.VALUE_INTEGER,
+                (v) -> v,
+                o -> defaultConverter(o, Integer.class, (v) -> Integer.parseInt(v.toString())));
+
+        public static final DataType<Long> LONG = new DataType<>(
+                Long.class.getName(),
+                Schema.ConceptProperty.VALUE_LONG,
+                (v) -> v,
+                o -> defaultConverter(o, Long.class, (v) -> Long.parseLong(v.toString())));
+
+        public static final DataType<Double> DOUBLE = new DataType<>(
+                Double.class.getName(),
+                Schema.ConceptProperty.VALUE_DOUBLE,
+                (v) -> v,
+                o -> defaultConverter(o, Double.class, (v) -> Double.parseDouble(v.toString())));
+
+        public static final DataType<Float> FLOAT = new DataType<>(
+                Float.class.getName(),
+                Schema.ConceptProperty.VALUE_FLOAT,
+                (v) -> v,
+                o -> defaultConverter(o, Float.class, (v) -> Float.parseFloat(v.toString())));
+
+        public static final DataType<LocalDateTime> DATE = new DataType<>(
+                LocalDateTime.class.getName(),
+                Schema.ConceptProperty.VALUE_DATE,
+                (d) -> d.atZone(ZoneId.systemDefault()).toInstant().toEpochMilli(),
+                (o) -> {
+                    if (o == null) return null;
+                    if (!(o instanceof Long)) {
+                        throw new InvalidConceptValueException(ErrorMessage.INVALID_DATATYPE.getMessage(o, Long.class.getName()));
+                    }
+                    return LocalDateTime.ofInstant(Instant.ofEpochMilli((long) o), ZoneId.systemDefault());
+                });
```

Comment:
At this point, I would make these classes implementing an interface:

```
private static DateType implements DataType<LocalDateTime> {
    @Override
    public D getValue(Object o) {
        if (o == null) return null;
        if (!(o instanceof Long)) {
            throw new InvalidConceptValueException(ErrorMessage.INVALID_DATATYPE.getMessage(o, Long.class.getName()));
        }
        return LocalDateTime.ofInstant(Instant.ofEpochMilli((long) o), ZoneId.systemDefault());
    }
    ... etc ...
}
```
You can also give the interface a default implementation for the converter functions, to save restating it.